### ISSUE TITLE
Install instructions for ctags

### ISSUE THREAD

**Issue by solosuper** [mariokonrad/marnav#36]
Hi,
Thanks for a great library.

What version of `ctags` are you using? Perhaps a comment about required version and a link to installation instructions could be provided?

---
#### Details
I'm trying to build the documentation using `make doc` on Ubuntu 18.04. The first error is

```
CMake Error at cmake/Documentation.cmake:20 (add_dependencies):
  The dependency target "tags" of target "doc" does not exist.
```
I find that the `tags` target is created by `CTags.cmake`, and the file indicates that i might be missing `ctags` and/or `cscope`

```cmake
find_program(CTAGS_PATH ctags)
find_program(CSCOPE_PATH cscope)

if(CTAGS_PATH AND CSCOPE_PATH)
	message(STATUS "Found ctags: ${CTAGS_PATH}")
	message(STATUS "Found cscope: ${CSCOPE_PATH}")
```
, so I tried installing it.

```bash
sudo apt install ctags cscope
```
This gives me "Exuberant Ctags 5.9", but `make tags` (or `make doc`) outputs the error
```
ctags: Unknown option: --extras
```

Including `doxygen` and the proper `ctags` and `cscope` in your docker images could be a sufficient solution for me personally, as I'm only looking for a way to generate the documentation.

**Comment by solosuper** [comment]
Replacing `--extras` with `--extra` solves the problem on Ubuntu 18.04

https://github.com/mariokonrad/marnav/blob/master/cmake/CTags.cmake#L51

**Comment by mariokonrad** [comment]
In the past, I used the one provided by Ubuntu 18.04 as you described. Then I moved to https://github.com/universal-ctags/ctags, and currently I'm using more or less HEAD. CTags.cmake changed in 685c41870bdfea347fd3ed760992fb588376cbf3 (2019-06-21) to use `--extras` instead of `--extra`. I was not aware that someone else also uses it, sorry.

I'm using vim (and gvim) for development, some time after the commit above I switched to the plugin _gutentags_, and I haven't used bare ctags again ever since.

I honestly don't know how useful it still is. Are you interested in keeping this feature (of the build)?
If yes, CTags.cmake could check `ctags` for options and do the right thing, not a hard thing to do actually.

**Comment by mariokonrad** [comment]
The detection could be solved for example like: https://github.com/mariokonrad/marnav/blob/ctags-check/cmake/CTags.cmake#L41-L56

**Comment by solosuper** [comment]
For me, no, not interested : ) 

Honestly, at first I thought it was some other way of generating [tag files for doxygen](https://www.doxygen.nl/manual/external.html), so that other people can include your tagfile in their doxyfile to get clickable links to marnav datatypes.

I fear I should have read more about ctags and tag files in order to not ask stupid questions, but I could not really see why the `doc` target should depend on the `tags` target. I've happily generated lots of doxygen documentation without it : )

**Comment by solosuper** [comment]
> The detection could be solved for example like: https://github.com/mariokonrad/marnav/blob/ctags-check/cmake/CTags.cmake#L41-L56

Sure, that would do it :+1:

**Comment by mariokonrad** [comment]
I'd say the issue is resolved with 6b04ad9725daade13282b1d452399773f226b441
